lib/generate-doc.js

```diff
@@ -5,69 +5,108 @@ const { get: getSchema, set: setSchema } = require('./layer-schema')
-          }
-
-          // Reformat the path
-          keys[k.name] = '{' + k.name + '}'
+          const params = routeLayer.keys.map(k => {
+            let param
+            if (schema.parameters) {
+              param = schema.parameters.find(
+                p => p.name === k.name && p.in === 'path'
+              )
+            }
 
-          return Object.assign({
-            name: k.name,
-            in: 'path',
-            required: !k.optional,
-            schema: { type: 'string' }
-          }, param || {})
-        })
+            // Reformat the path
+            keys[k.name] = '{' + k.name + '}'
 
-        if (schema.parameters) {
-          schema.parameters.forEach((p) => {
-            if (!params.find((pp) => p.name === pp.name)) {
-              params.push(p)
-            }
+            return Object.assign(
+              {
+                name: k.name,
+                in: 'path',
+                required: !k.optional,
+                schema: { type: 'string' }
+              },
+              param || {}
+            )
           })
-        }
 
-        operation.parameters = params
-        path = pathToRegexp.compile(path)(keys, { encode: (value) => value })
-      }
+          if (schema.parameters) {
+            schema.parameters.forEach(p => {
+              if (!params.find(pp => p.name === pp.name)) {
+                params.push(p)
+              }
+            })
+          }
+
+          operation.parameters = params
+          path = pathToRegexp.compile(path)(keys, { encode: value => value })
+        }
 
-      doc.paths[path] = doc.paths[path] || {}
-      doc.paths[path][layer.method] = operation
-      setSchema(layer.handle, operation)
+        doc.paths[path] = doc.paths[path] || {}
+        doc.paths[path][layer.method] = operation
+        setSchema(layer.handle, operation)
+      })
     })
-  })
 
   return doc
 }
 
 function iterateStack (path, routeLayer, layer, cb) {
   cb(path, routeLayer, layer)
+
+  // handle nested routes
+  if (layer.name === 'router') {
+    layer.handle.stack.forEach(l => {
+      path = path || ''
+      iterateStack(path + split(layer.regexp).join('/'), layer, l, cb)
+    })
+  }
+
   if (!layer.route) {
     return
   }
-  layer.route.stack.forEach((l) => iterateStack(path + layer.route.path, layer, l, cb))
+  layer.route.stack.forEach(l =>
+    iterateStack(path + layer.route.path, layer, l, cb)
+  )
+}
+
+// https://github.com/expressjs/express/issues/3308#issuecomment-300957572
+function split (thing) {
+  if (typeof thing === 'string') {
+    return thing.split('/')
+  } else if (thing.fast_slash) {
+    return []
+  } else {
+    var match = thing
+      .toString()
+      .replace('\\/?', '')
+      .replace('(?=\\/|$)', '$')
+      .match(/^\/\^((?:\\[.*+?^${}()|[\]\\/]|[^.*+?^${}()|[\]\\/])*)\$\//)
+    return match
+      ? match[1].replace(/\\(.)/g, '$1').split('/')
+      : '<complex:' + thing.toString() + '>'
```

Comment:
In this case generate-doc blows up because `'<complex:' + thing.toString() + '>'.join()` is not a function